packages/flutter/lib/src/material/refresh_indicator.dart

```diff
@@ -211,6 +232,12 @@ class RefreshIndicator extends StatefulWidget {
   /// [Future] must complete when the refresh operation is finished.
   final RefreshCallback onRefresh;
 
+  /// A function used to get current status of [RefreshIndicator] to update UI
+  /// as required while using [RefreshIndicator.noSpinner].
+  ///
+  /// Returns value of these enums as string: [_RefreshIndicatorMode]
+  final void Function(String? mode)? onModeChange;
```

Comment:
If we're going to do this then the mode enum should be public and passed along here, rather than the string representation of a private type. We'll also need to explain what `mode == null` means (hopefully that's not really needed).

Reply:
would we want to rename `RefreshIndicatorMode` to `RefreshIndicatorStatus` as it seems more reasonable. Mode is something that doesn't change much during execution while status is a real-time indicator of something.

What do you think?

Reply:
`RefreshIndicatorMode` was used by `refresh.dart` of Cupertino, this is again the initial reason of enum being private.

Hence I've renamed it as above.